src/main/java/uk/nhs/digital/nhsconnect/nhais/service/edifact_to_fhir/AbstractForenameTransactionMapper.java

```diff
@@ -0,0 +1,17 @@
+package uk.nhs.digital.nhsconnect.nhais.service.edifact_to_fhir;
+
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchOperation;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.AmendmentPatchRemoval;
+import uk.nhs.digital.nhsconnect.nhais.model.jsonpatch.JsonPatches;
+
+public abstract class AbstractForenameTransactionMapper {
```

Comment:
The remove character % is only permitted in the first forename. You don't need to implement this for second and third.

Reply:
that simplifies logic :)

Reply:
@grzegorz-robak, if removal can be only for first forename, then I remove this distinct